### DISCUSSION TITLE
vLLM failing to recognize GPU from latest official docker image

### DISCUSSION THREAD

**Discussion by DrappierTechnologies** [vllm-project/vllm#5661] [Q&A]
Hey folks,

I figured I'd post this here before opening a ticket to see if anyone has encountered this and its just a configuration issue on my part. I have been running vLLM's latest, officially published, docker image within my project up until about a couple months back. I got busy with other stuff and recently came back to try and work on the same project. When I run the latest image now it throws the following error:

`2024-06-18 21:24:16   return torch._C._cuda_getDeviceCount() > 0
2024-06-18 21:24:16 Traceback (most recent call last):
2024-06-18 21:24:16   File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
2024-06-18 21:24:16     return _run_code(code, main_globals, None,
2024-06-18 21:24:16   File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
2024-06-18 21:24:16     exec(code, run_globals)
2024-06-18 21:24:16   File "/workspace/vllm/entrypoints/openai/api_server.py", line 236, in <module>
2024-06-18 21:24:16     engine = AsyncLLMEngine.from_engine_args(engine_args)
2024-06-18 21:24:16   File "/workspace/vllm/engine/async_llm_engine.py", line 622, in from_engine_args
2024-06-18 21:24:16     engine_configs = engine_args.create_engine_configs()
2024-06-18 21:24:16   File "/workspace/vllm/engine/arg_utils.py", line 286, in create_engine_configs
2024-06-18 21:24:16     device_config = DeviceConfig(self.device)
2024-06-18 21:24:16   File "/workspace/vllm/config.py", line 496, in __init__
2024-06-18 21:24:16     raise RuntimeError("No supported device detected.")
2024-06-18 21:24:16 RuntimeError: No supported device detected.`

Not sure if something has changed in the base image or a dependent library but I couldn't find anything recent related to this error in the docs, issues, or discussions. Any insight would be very much appreciated.

Docker compose:
`  vllm:
    image: vllm/vllm-openai:latest
    command: ["--model", "TheBloke/Mistral-7B-Instruct-v0.2-code-ft-GPTQ", "--quantization", "gptq", "--dtype", "float16", "--revision", "gptq-4bit-32g-actorder_True", "--tokenizer", "TheBloke/Mistral-7B-Instruct-v0.2-code-ft-GPTQ", "--tokenizer-revision", "gptq-4bit-32g-actorder_True", "--max-num-batched-tokens", "6000", "--max-model-len", "6000"]
    restart: always
    shm_size: '32gb'
    ports:
        - "9002:8000"
    networks:
        - portfoliotracker_network
    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              count: all
              capabilities: [gpu]`

**Comment by elkay** [comment]
I was in the same situation, got distracted with other stuff the past 4-6 months and looping back now. Latest is also giving me the same error. Did you ever resolve this?

**Comment by elkay** [comment]
I've actually tried going back to older images (5.x, 4.x, 3.x, 2.x) and get the same crash. nvidia-smi shows correctly as follows:

`C:\AI-Content\vllm-docker>nvidia-smi
Thu Sep 12 12:25:28 2024
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 561.09                 Driver Version: 561.09         CUDA Version: 12.6     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                  Driver-Model | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 4090      WDDM  |   00000000:01:00.0  On |                  Off |
|  0%   48C    P0             55W /  400W |    2420MiB /  24564MiB |      6%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+

+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI        PID   Type   Process name                              GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|    0   N/A  N/A      2596    C+G   ...siveControlPanel\SystemSettings.exe      N/A      |
|    0   N/A  N/A      6044    C+G   C:\Windows\explorer.exe                     N/A      |
|    0   N/A  N/A      7320    C+G   ...aam7r\AcrobatNotificationClient.exe      N/A      |
|    0   N/A  N/A      8436    C+G   ...al\Discord\app-1.0.9162\Discord.exe      N/A      |
|    0   N/A  N/A      8708    C+G   ...sair iCUE5 Software\QmlRenderer.exe      N/A      |
|    0   N/A  N/A      9176    C+G   ...s\System32\Kinect\KinectService.exe      N/A      |
|    0   N/A  N/A      9364    C+G   ...les (x86)\Battle.net\Battle.net.exe      N/A      |
|    0   N/A  N/A     11364    C+G   ...nt.CBS_cw5n1h2txyewy\SearchHost.exe      N/A      |
|    0   N/A  N/A     11496    C+G   ...2txyewy\StartMenuExperienceHost.exe      N/A      |
|    0   N/A  N/A     13936    C+G   ...5n1h2txyewy\ShellExperienceHost.exe      N/A      |
|    0   N/A  N/A     14744    C+G   ...oogle\Chrome\Application\chrome.exe      N/A      |
|    0   N/A  N/A     14936    C+G   ...t.LockApp_cw5n1h2txyewy\LockApp.exe      N/A      |
|    0   N/A  N/A     18720    C+G   ...ekyb3d8bbwe\PhoneExperienceHost.exe      N/A      |
|    0   N/A  N/A     18900    C+G   ...\Docker\frontend\Docker Desktop.exe      N/A      |
|    0   N/A  N/A     19476    C+G   ...CBS_cw5n1h2txyewy\TextInputHost.exe      N/A      |
|    0   N/A  N/A     19644    C+G   ...US\ArmouryDevice\asus_framework.exe      N/A      |
|    0   N/A  N/A     23580    C+G   ...air\Corsair iCUE5 Software\iCUE.exe      N/A      |
|    0   N/A  N/A     24500    C+G   ...ecurityApp\MicrosoftSecurityApp.exe      N/A      |
|    0   N/A  N/A     33296    C+G   ...__8wekyb3d8bbwe\WindowsTerminal.exe      N/A      |
+-----------------------------------------------------------------------------------------+`

**Comment by elkay** [comment]
FYI I was able to fix this. Turns out that my Docker Desktop itself was out of date and installing the latest update fixed this error. Was likely due to too new a video driver with an older Docker Desktop.

**Comment by NickCheng0921** [comment]
Adding that on Windows WSL + Docker Desktop, there's a flag that needs to be set under
Settings -> Enable integration with my default WSL distro -> check
on Docker Desktop version 4.76.0

otherwise GPU access is disabled

You can check whether your docker desktop sees the GPU by running
docker run --rm --gpus all nvidia/cuda:13.0.0-base-ubuntu22.04 nvidia-smi
if the table comes up, GPU access is enabled
